pkg/storage/internal/actions/abort.go

```diff
@@ -0,0 +1,88 @@
+package actions
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/logging"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/storage/entity"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/storage/history"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/wdk"
+)
+
+type abortAction struct {
+	logger           *slog.Logger
+	transactionsRepo TransactionsRepo
+}
+
+func newAbortAction(logger *slog.Logger, transactions TransactionsRepo) *abortAction {
+	return &abortAction{
+		logger:           logging.Child(logger, "abort_action"),
+		transactionsRepo: transactions,
+	}
+}
+
+func (a *abortAction) AbortAction(ctx context.Context, userID int, args *wdk.AbortActionArgs) (*wdk.AbortActionResult, error) {
+	txEntity, err := a.transactionsRepo.FindUniqueTransactionByReference(ctx, userID, *args.Reference)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find unique transaction by reference: %w", err)
+	}
+
+	if txEntity == nil && len(*args.Reference) == 64 {
```

Comment:
It seems like magic - IMHO we should provide some `NOTE` comment or separate this logic to another function (also with the comment)

Reply:
I would solve this with extracting a method to use here like
```suggestion
if txEntity == nil && a.isPotentiallyTxID(args.Reference) {
```

Reply:
method extracted